core/fuzzer.h

```diff
@@ -6,16 +6,23 @@
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wunused-parameter"
 #pragma GCC diagnostic ignored "-Wpedantic"
+#include "backends/p4tools/common/control_plane/bfruntime/bfruntime.pb.h"
 #include "p4/v1/p4runtime.pb.h"
 #pragma GCC diagnostic pop
 
 namespace P4Tools::RTSmith {
 
-using InitialP4RuntimeConfig = std::vector<p4::v1::WriteRequest>;
-using P4RuntimeUpdateSeries = std::vector<std::pair<uint64_t, p4::v1::WriteRequest>>;
+enum class FuzzerType { P4Runtime, BFRuntime };
 
-class P4RuntimeFuzzer {
+using ProtobufMessagePtr = std::unique_ptr<google::protobuf::Message>;
+using InitialConfig = std::vector<ProtobufMessagePtr>;
+using UpdateSeries = std::vector<std::pair<uint64_t, ProtobufMessagePtr>>;
+
+class RuntimeFuzzer {
  private:
+    /// fuzzer type to distinguish between P4Runtime and BFRuntime fuzzers.
+    FuzzerType fuzzerType;
```

Comment:
Personally, I would instead implement a new Fuzzer class and share common code in a common fuzzer. So, if the requested type is BFRuntime, generate a BFRuntimeFuzzer instead. Not every P4 target supports BFruntime anyway, it is mostly only Tofino so we can also move the code there.

Reply:
> Personally, I would instead implement a new Fuzzer class and share common code in a common fuzzer. So, if the requested type is BFRuntime, generate a BFRuntimeFuzzer instead.

Yeah we can remove the `fuzzerType` here. And a subclass `BFRuntimeFuzzer` is what we do now. 

> we can also move the code there.

So keeping P4RuntimeFuzzer in "core/fuzzer", and moving the contents of `BFRuntimeFuzzer` to "targets/tofino/TofinoTnaFuzzer", right?

Reply:
Yes!